webflow-hacks/form-hacks/29/29.ts

```diff
@@ -0,0 +1,22 @@
+// when DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  // declare constant selectors
+  const FORM_SELECTOR = '[fs-hacks-element="hack29-form"]';
+  const NAME_INPUT_SELECTOR = '[fs-hacks-element="hack29-name-input"]';
+  const MESSAGE_SELECTOR = '.w-form-done';
+  const form = document.querySelector<HTMLFormElement>(FORM_SELECTOR);
+  // early return
+  if (!form) return;
+  const nameInput = form.querySelector<HTMLInputElement>(NAME_INPUT_SELECTOR);
+  const messageDiv = form.querySelector<HTMLDivElement>(MESSAGE_SELECTOR);
+  // when form is submitted
+  form.addEventListener('submit', function () {
+    if (!nameInput || !messageDiv) return;
```

Comment:
Same here, why add the return inside the listener when you can directly stop the listener from ever being added?
You should put this return outside the listener

Reply:
done